Service/FindOrganizationThroughRepositoriesService.php

```diff
@@ -8,42 +8,148 @@
-        $repository->setValue('issue_open_count', $github['issue_open_count']);
-        $repository->setValue('programming_languages', $github['programming_languages']);
+        $this->io = $io;
 
+        return $this;
+    }
+
+    
+
+    /**
+     * This function gets the content of the given url.
+     *
+     * @param string      $url
+     * @param string|null $path
+     *
+     * @return array|null
+     */
+    public function requestFromUrl(string $url, ?string $path = null): ?array
+    {
+        // @TODO callService needs to be used without source, full domain here is variable
+        // if ($path !== null) {
+        //     $parse = parse_url($url);
+        //     $url = str_replace([$path], '', $parse['path']);
+        // }
+
+        // if ($response = $this->callService->call('GET', $url)) {
+        //     return json_decode($response->getBody()->getContents(), true);
+        // }
+
+        return null;
+    }
+
+    /**
+     * This function gets all the github repository details.
+     *
+     * @param array $item a repository from github with a publicclode.yaml file
+     *
+     * @return array
+     */
+    public function getGithubRepositoryInfo(array $item): array
+    {
+        return [
+            'source'                  => 'github',
+            'name'                    => $item['name'],
+            'url'                     => $item['html_url'],
+            'avatar_url'              => $item['owner']['avatar_url'],
+            'last_change'             => $item['updated_at'],
+            'stars'                   => $item['stargazers_count'],
+            'fork_count'              => $item['forks_count'],
+            'issue_open_count'        => $item['open_issues_count'],
+            //            'merge_request_open_count'   => $this->requestFromUrl($item['merge_request_open_count']),
+            'programming_languages'   => $this->requestFromUrl($item['languages_url']),
+        //    'organisation'            => $item['owner']['type'] === 'Organization' ? $this->getGithubOwnerInfo($item) : null,
+            //            'topics' => $this->requestFromUrl($item['topics'], '{/name}'),
+            //                'related_apis' => //
+        ];
+    }
+
+    /**
+     * This function is searching for repositories containing a publiccode.yaml file.
+     *
+     * @param string $slug endpoint to request
+     *
+     * @throws GuzzleException
+     *
+     * @return array|null|Response
+     */
+    public function getRepositoryFromUrl(string $slug)
+    {
+        try {
+            $response = $this->callService->call($this->githubApi, '/repos/'.$slug);
+        } catch (Exception $e) {
+            // @TODO Monolog ?
+            isset($this->io) && $this->io->error("Error found trying to fetch '/repos/'.$slug : " . $e->getMessage());
+            return null;
+        }
+
+        $response = $this->callService->decodeResponse($this->githubApi, $response, 'application/json');
```

Comment:
Is die derde optie nodig?

Reply:
Ja anders probeert de callservice vanuit een request de content type uit te vogelen